assets/js/script.js

```diff
@@ -22,14 +25,171 @@ const topFunction = () => {
+};
+
+const closeDetails = () => {
+  if (hasClass(pdContainer, 'show')) {
+    pdContainer.classList.remove('show');
+  }
+};
+
+// 4. form-validation
+const inputName = document.getElementById('userName');
+const inputEmail = document.getElementById('userEmail');
+const form = document.getElementById('form-contact');
+const textArea = document.getElementById('contact-message');
+
+const isRequired = (value) => (value !== '');
+const isBetween = (length, min, max) => (!(length < min || length > max));
+const isEmailValid = (email) => {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(email);
+};
+
+const showError = (input, message) => {
+  const formField = input.parentElement;
+  formField.classList.remove('success');
+  formField.classList.add('error');
+
+  const error = formField.querySelector('small');
+  error.textContent = message;
+};
+
+const showSuccess = (input) => {
+  const formField = input.parentElement;
+  formField.classList.remove('error');
+  formField.classList.add('success');
+
+  const error = formField.querySelector('small');
+  error.textContent = '';
+};
+
+const checkName = () => {
+  const username = inputName.value.trim();
+  let valid = false;
+  const min = 3;
+  const max = 30;
+  if (!isRequired(username)) {
+    showError(inputName, 'Username cannot be blank');
+  } else if (!isBetween(username.length, min, max)) {
+    showError(inputName, 'User name must be between 3 and 30 characters');
+  } else {
+    showSuccess(inputName);
+    valid = true;
+  }
+  return valid;
+};
+
+const checkEmail = () => {
+  let valid = false;
+  const email = inputEmail.value.trim();
+
+  const pattern = /[A-Z]/;
+  if (pattern.test(inputEmail.value)) {
+    showError(inputEmail, 'Email has to lower case');
+  } else if (!isRequired(email)) {
+    showError(inputEmail, 'Please fill email');
+  } else if (!isEmailValid(email)) {
+    showError(inputEmail, 'Please provide a valid email');
+  } else {
+    showSuccess(inputEmail);
+    valid = true;
+  }
+  return valid;
+};
+
+const checkMessage = () => {
+  let valid = false;
+  const min = 15;
+  const max = 500;
+  const textMessage = textArea.value.trim();
+
+  if (!isRequired(textMessage)) {
+    showError(textArea, 'Please enter message');
+  } else if (!isBetween(textMessage.length, min, max)) {
+    showError(textArea, 'Message should at least 15 - 500 characters');
+  } else {
+    showSuccess(textArea);
+    valid = true;
+  }
+  return true;
+};
+
+function val() {
+  return true;
+}
+
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  const isUsernameValid = checkName();
+  const isEmailValid = checkEmail();
+  const isTextFieldValid = checkMessage();
+  const isFormValid = isUsernameValid && isEmailValid && isTextFieldValid;
+
+  if (isFormValid) {
+    val();
+  }
```

Comment:
- [ ] [OPTIONAL] Hi, your code functionality is working well, however, this function call here seems completely unnecessary. the `val` function simply returns `true` and you never use that value so removing this doesn't seem like it will have any effect on your code.